Replace start & end dates in course run list with modified date

dsjen opened the pull request:
# Original
![in-dev-original](https://user-images.githubusercontent.com/5265058/40450606-33babdf2-5eaa-11e8-8c40-11b00cb642ec.png)

![in-preview-original](https://user-images.githubusercontent.com/5265058/40450608-33cff3e8-5eaa-11e8-921c-fd8a56c09711.png)

![published-original](https://user-images.githubusercontent.com/5265058/40450609-33e065b6-5eaa-11e8-9995-fdc823ee59ac.png)

**(I don't have a screenshot of the original "Studio Request" tab, but it just had a Start date)**

# New
![in-dev-new](https://user-images.githubusercontent.com/5265058/40450610-33f1e3cc-5eaa-11e8-9a37-7ba86a58c4a6.png)

![in-preview-new](https://user-images.githubusercontent.com/5265058/40450611-33ffe508-5eaa-11e8-86c7-19e334893acf.png)

![published-new](https://user-images.githubusercontent.com/5265058/40450612-341418fc-5eaa-11e8-83c5-0ac8b9732194.png)

![studio-request-new](https://user-images.githubusercontent.com/5265058/40450613-3420cae8-5eaa-11e8-99c4-24230e6233e4.png)

@edx/educator-dahlia @sstack22

fysheets commented:
Is the course run info (eg 3T2018b) going to be added as well? Or was that not available?

dsjen commented:
@fysheets -- thanks for asking! It turns out that the run info isn't available in the `CourseRun` model is calculated instead. It seemed like more effort than it was worth to add it, and I checked in with @sstack22 who agreed.

dsjen commented:
I've updated the translations an the build is passing now. I also added documentation to the README on how to extract the translations and check them in for your build to pass.

fysheets commented on README.rst:
```diff
@@ -25,6 +25,14 @@ How To Contribute
+
+Is the build failing because translations are out of date?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Run ``make check_translations_up_to_date`` and check in the generated *.mo & *.po files to your PR.
```
👍
